opengovsg/FormSG
# feat(api): collapse spcp/myinfo redirect endpoint to new endpoint

Part 1 of #1517 

## Problem
Previously, our `/auth/redirect` endpoints were split between the `MyInfo` and `spcp` sections, which made code harder to maintain and our frontend had separated logic. 

Separately, this is also part of the `/api/v3` migration to shift endpoints to their new homes

## Solution
This PR merges the 2 endpoints into 1 and shifts it into a new route rooted at `/api/v3`

**Features**:

1. Created a new controller method under `auth.controller` to handle the redirects for backend
2. Created a new router to service the new `/api/v3` routes
3. New `AuthService` on frontend side as part of react migration 

**Notes**
1. there is no decision making on the frontend now. instead, decisions are deferred to the backend and the frontend just transparently sends across its data
2. `MyInfoService` will be fully deleted in part 2 of this PR. it is left as is first. 

## Tests
Tests are [here](https://github.com/opengovsg/FormSG/pull/1685) to avoid review fatigue

## Manual Tests
1. Create a SP form. Click on activate and go to the form. Click on keep me logged in. Clicking on sign in with singpass should redirect you to the Singpass page. Ensure that after signing in, refreshing the page still keeps you signed in.
2. Create a SP form. Click on activate and go to the form. Clicking on sign in with singpass should redirect you to the Singpass page.
3. Create a email mode form with authentication set to myInfo. Click on activate and go to the form. Clicking on sign in with singpass should redirect to the singpass page.
